src/main/java/seedu/address/model/event/EventTime.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * This is an EventTime class that represents the Time of an Event.
+ */
+public class EventTime {
+    public static final String MESSAGE_CONSTRAINTS = "Time should be in 2359 format.";
+
+    public final LocalTime time;
+
+    /**
+     * Constructors an {@code EventTime}
+     *
+     * @param time    A valid time.
+     */
+    public EventTime(String time) {
+        requireNonNull(time);
+        checkArgument(isValidTime(time), MESSAGE_CONSTRAINTS);
+        this.time = LocalTime.parse(time, DateTimeFormatter.ofPattern("HHmm"));
+    }
+
+    public EventTime() {
+        this.time = LocalTime.MIN;
+    }
+
+    /**
+     * Returns true if a given string is a valid time.
+     *
+     * @param test A String to be tested if it is a valid time.
+     * @return A boolean indicating if the string is a valid time.
+     */
+    public static boolean isValidTime(String test) {
+        // There must be a better way to do this.
+        try {
+            LocalTime.parse(test, DateTimeFormatter.ofPattern("HHmm"));
+            return true;
```

Comment:
Would this part count as you using try-catch as the logic for isValidTime?

Reply:
Might look a bit better with something like
````
boolean isValid;
try {
  LocalTime.parse(...
  isValid = true;
} catch (DateTimeParseException e) {
  isValid = false;
}
return isValid;
````

This or we can go with the SimpleDateFormat solution you suggested, which might look even better.